src/components/percentage/percentage.jsx

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+} from 'react';
+import FormControl from '../form-control/form-control.jsx';
+import FormControlIcons from '../form-control-icons/form-control-icons.jsx';
+import styles from './percentage.css';
+import useMounted from '../../hooks/use-mounted.js';
+import useValidation from '../../hooks/use-validation.jsx';
+
+function getClassName(className, validationMessage) {
+  if (className) return className;
+  return validationMessage ? styles['invalid-input'] : styles.input;
+}
+
+const Percentage = forwardRef(function Percentage(props, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
+  const inputRef = useRef();
+  const mounted = useMounted();
+  const [validationMessage, validate] = useValidation(props.validationMessage, inputRef);
+
+  function onBlur(event) {
+    validate();
+    props.onBlur(event);
+  }
+
+  function onChange(event) {
+    validate();
+    props.onChange(event);
+  }
+
+  useEffect(() => {
+    if (!mounted.current) return;
+
+    // The MDS Input is using an uncontrolled `<input>`.
+    // In order to set a new provided value prop, we
+    // set the internal state of the `<input>`.
+    inputRef.current.value = props.value || '';
+  }, [props.value]);
+
+  useImperativeHandle(ref, () => ({
+    get dirty() {
+      const providedValue = props.value || undefined;
+      return !Object.is(providedValue, this.value);
+    },
+    id: props.id,
+    name: props.name,
+    get value() {
+      return parseFloat(inputRef.current.value) || undefined;
+    },
+  }));
+
+  return (
+    <FormControl
+      className={props.cssContainer}
+      error={validationMessage}
+      id={props.id}
+      label={props.label}
+      readOnly={props.readOnly}
+      required={props.required}
+    >
+      <input
+        aria-describedby={`${props.id}Hint`}
+        autoFocus={props.autoFocus} // eslint-disable-line jsx-a11y/no-autofocus
+        className={getClassName(props.className, validationMessage)}
+        defaultValue={props.value}
+        id={props.id}
+        max={100}
+        min={0}
+        step={0.01}
+        name={props.name}
+        onBlur={onBlur}
+        onChange={onChange}
+        onFocus={props.onFocus}
```

Comment:
Do we have a use-case for this event? Or is it some foreseeable use-case? (Same for `onInput`, `onKeyDown`, `props.onBlur`)

Reply:
A lot of these are me trying to maintain functional parity with our other `input`-based components. I can remove the stuff we don't currently use, if you like.

Reply:
I did remove all of these except for `onChange`.

Reply:
Yeah, I'm not entirely convinced we need a large events API. Or if we do then it should come with some design UX feedback, like: we want our form fields to auto-save on blur or something.